core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1685,6 +1685,7 @@ public FetchDataInfo read(RemoteStorageFetchInfo remoteStorageFetchInfo) throws
                 return new FetchDataInfo(new LogOffsetMetadata(offset), MemoryRecords.EMPTY);
             }
 
+            startPos = lookupPositionForOffset(remoteLogSegmentMetadata, firstBatch.baseOffset());
```

Comment:
offset-index is a sparse one, so the position returned here might be same as the previous one.